rust/lance-core/src/io/writer/statistics.rs

```diff
@@ -239,11 +265,9 @@ fn get_string_statistics<T: OffsetSizeTrait>(arrays: &[&ArrayRef]) -> Statistics
 
         array.iter().for_each(|value| {
             if let Some(mut val) = value {
-                // Truncate string to the longest length that is still a valid UTF8 string,
-                // while being of BINARY_PREFIX_LENGTH lenght or just greate. This is to avoid
-                // comparing potentially very long strings.
-                if val.len() > BINARY_PREFIX_LENGTH + 4 {
-                    val = truncate_utf8(val, BINARY_PREFIX_LENGTH + 4).unwrap();
+                if val.len() > BINARY_PREFIX_LENGTH {
+                    val = truncate_utf8(val, BINARY_PREFIX_LENGTH).unwrap();
+                    bounds_truncated = true;
```

Comment:
It turns out we needed to add 4 here so we included the next unicode character (at most 4 bytes) so we can decide later whether we need to increment. However, I found it simpler to read if we used a boolean to track whether we have truncated.